src/api.rs

```diff
@@ -33,16 +33,110 @@ pub struct EncoderConfig {
   pub key_frame_interval: u64,
   pub low_latency: bool,
   pub quantizer: usize,
-  pub speed: usize,
-  pub tune: Tune
+  pub tune: Tune,
+  pub min_block_size: BlockSize,
+  pub multiref: bool,
+  pub fast_deblock: bool,
+  pub reduced_tx_set: bool,
+  pub tx_domain_distortion: bool,
+  pub encode_bottomup: bool,
+  pub rdo_tx_decision: bool,
+  pub prediction_modes: PredictionModesSetting,
+  pub include_near_mvs: bool,
 }
 
 impl Default for EncoderConfig {
   fn default() -> Self {
-    EncoderConfig { key_frame_interval: 30, low_latency: true, quantizer: 100, speed: 0, tune: Tune::Psnr  }
+    const DEFAULT_SPEED: usize = 3;
+    Self::with_speed_preset(DEFAULT_SPEED)
   }
 }
 
+impl EncoderConfig {
+  pub fn with_speed_preset(speed: usize) -> Self {
+    let mut cfg = EncoderConfig {
+      key_frame_interval: 30,
+      low_latency: true,
+      quantizer: 100,
+      tune: Tune::Psnr,
+      ..Default::default()
+    };
+    cfg.apply_speed_preset(speed);
+    cfg
+  }
+
+  pub fn apply_speed_preset(&mut self, speed: usize) {
```

Comment:
Yes! This is wonderful way to keep only once function to control various switch affecting the time complexity of encoder. Thanks for nice idea!